Library/Homebrew/extend/os/mac/extend/ENV/std.rb

```diff
@@ -72,15 +72,17 @@ def remove_macosxsdk(version = MacOS.version)
     remove "CMAKE_FRAMEWORK_PATH", "#{sdk}/System/Library/Frameworks"
   end
 
-  def macosxsdk(version = MacOS.version)
+  def macosxsdk(version = nil, formula: nil)
     # Sets all needed lib and include dirs to CFLAGS, CPPFLAGS, LDFLAGS.
     remove_macosxsdk
-    version = version.to_s
-    append_to_cflags("-mmacosx-version-min=#{version}")
+    append_to_cflags("-mmacosx-version-min=#{version || MacOS.version}")
     self["CPATH"] = "#{HOMEBREW_PREFIX}/include"
     prepend "LDFLAGS", "-L#{HOMEBREW_PREFIX}/lib"
 
-    return unless (sdk = MacOS.sdk_path_if_needed(version))
+    sdk = formula ? MacOS.sdk_for_formula(formula, version) : MacOS.sdk(version)
+    return unless MacOS.sdk_root_needed? || sdk.source == :xcode
```

Comment:
```suggestion
    return unless MacOS.sdk_root_needed?
    
    sdk = formula ? MacOS.sdk_for_formula(formula, version) : MacOS.sdk(version)
    return if sdk.source != :xcode
```

Reply:
This should probably be:

```rb
return if !MacOS.sdk_root_needed? && sdk.source != :xcode
```

or similar. If `MacOS.sdk_root_needed?` is false then we don't want to return right away until we check the SDK is not from Xcode.

Reply:
Yup, sounds good. As mentioned: this is why I hate `unless .. ||` 😁